### PR TITLE
Make surface contour highlights on hover toggleable 

### PR DESCRIPTION
**Pull request by etpinard**
@mdtusz @cldougl 

### In brief

This PR:

- makes `surface` contour highlights on hover toggleable through the `contours.?.highlight` attribute (we weren't updating the correct [`gl-surface3d`](https://github.com/gl-vis/gl-surface3d/blob/master/surface.js) property).
- fixes a typo (or maybe a careless mistake) in the surface attributes: `highlightColor` is now `highlightcolor` and `highlightWidth` is now `highlightwidth`. A backward compatibility block is added.
- makes the surface contours line and scene spikes use `Color.defaultLine` as default (like everybody else :smiley: ).
- adds several missing `surface` attribute descriptions.

### CODE REVIEW

File: src/traces/surface/attributes.js [hunk 1]
```diff
@@ -20,8 +20,12 @@ function makeContourProjAttr(axLetter) {
         role: 'info',
         dflt: false,
         description: [
-            'Sets whether or not the dynamic contours are projected',
-            'along the', axLetter, 'axis.'
+            'Determines whether or not these contour lines are projected',
```
Comment (etpinard) [inline src/traces/surface/attributes.js]: @cldougl (or someone else from the doc team) could you review :arrow_down: ?

File: src/traces/surface/attributes.js [hunk 2]
```diff
@@ -44,7 +45,7 @@ function makeContourAttr(axLetter) {
         color: {
             valType: 'color',
             role: 'style',
-            dflt: '#000'
+            dflt: Color.defaultLine,
```
Comment (etpinard) [inline src/traces/surface/attributes.js]: This required updating on baseline image :arrow_down:.

File: src/traces/surface/attributes.js (hunk 1 above)
Comment (cldougl) [inline src/traces/surface/attributes.js]: Clear to me!

File: src/traces/surface/attributes.js [hunk 3]
```diff
@@ -118,8 +138,15 @@ module.exports = {
...
         valType: 'boolean',
         role: 'info',
-        dflt: false
+        dflt: false,
+        description: [
+            'Determines whether or not a surface is drawn.',
+            'For example, set `hidesurface` to *false*',
+            '`contours.x.show` to *true* and',
+            '`contours.y.show` to *true* to draw a wire frame plot.',
+        ].join(' ')
```
Comment (cldougl) [inline src/traces/surface/attributes.js]: 💎 description--thanks for adding that!

File: test/jasmine/tests/plot_api_test.js [hunk 4]
```diff
@@ -711,8 +711,12 @@ describe('Test plot api', function() {
                 }
             }];
 
+            spyOn(Plots.subplotsRegistry.gl3d, 'plot');
```
Comment (etpinard) [inline test/jasmine/tests/plot_api_test.js]: fyi @mdtusz @monfera quick and dirty trick to test `Plotly.plot` with gl traces in a way that works on CircleCI.

**Comment by mdtusz** [comment]
Looks good to me! Shame about the attribute name typo, but such is life. 

💃

**Pull request merged**